reconcile/aws_cloudwatch_log_retention/integration.py

```diff
@@ -64,21 +65,26 @@ def check_cloudwatch_log_group_tag(
     return log_group_list
 
 
+def create_aws_log_client(aws_acct: dict) -> tuple[list, AWSApi]:
+    settings = queries.get_secret_reader_settings()
+    accounts = queries.get_aws_accounts(uid=aws_acct.get("uid"))
+    awsapi = AWSApi(1, accounts, settings=settings, init_users=False)
```

Comment:
Please add a return statement here

Reply:
There is another issue, `create_awsapi_client` is called within a loop, so `queries.get_secret_reader_settings()` and `queries.get_aws_accounts(uid=aws_acct.get("uid"))` (not needed as we already have account object) will called per iteration, and `AWSApi` is not cleaned up afterwards.

I suggest to change the structure to create one `AWSApi` to use:

```python
def create_awsapi_client(accounts: list, thread_pool_size: int) -> AWSApi:
    settings = queries.get_secret_reader_settings()
    return AWSApi(thread_pool_size, accounts, settings=settings, init_users=False)
```

```python
def run(dry_run: bool, thread_pool_size: int) -> None:
    aws_accounts = get_aws_accounts(cleanup=True)
    with create_awsapi_client(aws_accounts) as awsapi:
        for aws_acct in aws_accounts:
           ...
```